src/dlmalloc.rs

```diff
@@ -4,12 +4,13 @@
 // The original source was written by Doug Lea and released to the public domain
 
 use core::cmp;
+use core::marker::PhantomData;
 use core::mem;
 use core::ptr;
 
-use sys;
+use System;
 
-pub struct Dlmalloc {
+pub struct Dlmalloc<S: System> {
```

Comment:
Please add a default type parameter so that existing users don't have to change (much). The default type should use the same selection mechanism as currently for `mod sys`.